packages/test/test-utils/src/testFluidObject.ts

```diff
@@ -167,19 +168,29 @@ export class TestFluidObjectFactory implements IFluidDataStoreFactory {
         }
 
         const runtimeClass = mixinRequestHandler(
-            async (request: IRequest) => {
-                const router = await routerP;
+            async (request: IRequest, rt: FluidDataStoreRuntime) => {
+                // TestFluidObject doesn't actually implement IFluidRouter, it just has a request method,
+                // so we don't try to discover it like FluidObject<IFluidRouter>. This factory knows that
+                // the entrypoint on the data stores it creates is an object of that type because it passed
+                // it in (see the call to new runtimeClass(...) below), so it can cast safely here.
+                const router: TestFluidObject = ((await rt.IFluidLoadable?.handle?.get()) as TestFluidObject);
```

Comment:
i'd rather we add the IFluidRouter to TestFluidObject, and do the right casting, rather than work around the bug in the type